1.26.0 - Plugin: Slabinfo - compilation warnings (expect argument)

### Bug report summary

While cross-compilation of Netdata 1.26.0, I noticed these warnings: 
https://gist.github.com/BKPepe/82f9a66e8f01e1eb0565b57baa182c1d

Almost the same for collectors/proc.plugin/sys_class_infiniband.c and  `collectors/proc.plugin/proc_pagetypeinfo.c`


##### OS / Environment
Linux super_cool_name 5.4.0-48-generic #52-Ubuntu SMP Thu Sep 10 10:58:49 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=20.04
/etc/lsb-release:DISTRIB_CODENAME=focal
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 20.04.1 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="20.04.1 LTS (Focal Fossa)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 20.04.1 LTS"
/etc/os-release:VERSION_ID="20.04"
/etc/os-release:HOME_URL="https://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="https://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
/etc/os-release:PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
/etc/os-release:VERSION_CODENAME=focal
/etc/os-release:UBUNTU_CODENAME=focal

##### Netdata version
netdata v1.26.0

##### Component Name
Plugins: slabinfo, proc

##### Steps To Reproduce

- Try to cross-compile it for OpenWrt


##### Expected behavior
There shouldn't be warnings.

Hi,

Yes, target CPU is 32 bit. I tried your patch and I see there are still some warnigns.

```
collectors/slabinfo.plugin/slabinfo.c: In function 'read_file_slabinfo':
collectors/slabinfo.plugin/slabinfo.c:194:23: warning: format '%lu' expects argument of type 'long unsigned int', but argument 4 has type 'size_t {aka unsigned int}' [-Wformat=]
             slabdebug("    Line %zu has only %lu words, skipping", l, procfile_linewords(ff,l));
                       ^
collectors/slabinfo.plugin/slabinfo.c:16:23: note: in definition of macro 'slabdebug'
     fprintf(stderr, ##args); \
                       ^~~~
```

--- a/collectors/perf.plugin/perf_plugin.c
+++ b/collectors/perf.plugin/perf_plugin.c
@@ -422,7 +422,7 @@ static int perf_collect() {
             }
         }
 
-        if(unlikely(debug)) fprintf(stderr, "perf.plugin: successfully read event id = %u, value = %lu\n", current_event->id, current_event->value);
+        if(unlikely(debug)) fprintf(stderr, "perf.plugin: successfully read event id = %u, value = %"PRIu64"\n", current_event->id, current_event->value);
     }
 
     if(unlikely(perf_events[EV_ID_CPU_CYCLES].value == prev_cpu_cycles_value))

--- a/collectors/slabinfo.plugin/slabinfo.c
+++ b/collectors/slabinfo.plugin/slabinfo.c
@@ -191,7 +191,7 @@ struct slabinfo *read_file_slabinfo() {
     slabdebug("   Read %lu lines from procfile", (unsigned long)lines);
     for(l = 2; l < lines; l++) {
         if (unlikely(procfile_linewords(ff, l) < 14)) {
-            slabdebug("    Line %lu has only %lu words, skipping", (unsigned long)l, procfile_linewords(ff,l));
+            slabdebug("    Line %zu has only %lu words, skipping", l, procfile_linewords(ff,l));
             continue;
         }
 
@@ -231,7 +231,7 @@ struct slabinfo *read_file_slabinfo() {
         else
             s->obj_filling = 0;
 
-        slabdebug("    Updated slab %s: %lu %lu %lu %lu %lu / %lu %lu %lu / %lu %lu %lu / %lu %lu %hhu",
+        slabdebug("    Updated slab %s: %"PRIu64" %"PRIu64" %"PRIu64" %"PRIu64" %"PRIu64" / %"PRIu64" %"PRIu64" %"PRIu64" / %"PRIu64" %"PRIu64" %"PRIu64" / %"PRIu64" %"PRIu64" %hhu",
             name, s->active_objs, s->num_objs, s->obj_size, s->obj_per_slab, s->pages_per_slab,
             s->tune_limit, s->tune_batchcnt, s->tune_shared_factor,
             s->data_active_slabs, s->data_num_slabs, s->data_shared_avail,
@@ -304,7 +304,7 @@ unsigned int do_slab_stats(int update_every) {
             , "slabmemory"
         );
         for (s = sactive; s; s = s->next) {
-            printf("SET %s = %lu\n"
+            printf("SET %s = %"PRIu64"\n"
                 , s->name
                 , s->mem_usage
             );
@@ -334,7 +334,7 @@ unsigned int do_slab_stats(int update_every) {
             , "slabwaste"
         );
         for (s = sactive; s; s = s->next) {
-            printf("SET %s = %lu\n"
+            printf("SET %s = %"PRIu64"\n"
                 , s->name
                 , s->mem_waste
             );

--- a/collectors/slabinfo.plugin/slabinfo.c
+++ b/collectors/slabinfo.plugin/slabinfo.c
@@ -163,12 +163,12 @@ struct slabinfo *read_file_slabinfo() {
     slabdebug("-> Reading procfile %s", PLUGIN_SLABINFO_PROCFILE);
 
     static procfile *ff = NULL;
-	static long slab_pagesize = 0;
+    static long slab_pagesize = 0;
 
-	if (unlikely(!slab_pagesize)) {
-		slab_pagesize = sysconf(_SC_PAGESIZE);
-		slabdebug("   Discovered pagesize: %ld", slab_pagesize);
-	}
+    if (unlikely(!slab_pagesize)) {
+        slab_pagesize = sysconf(_SC_PAGESIZE);
+        slabdebug("   Discovered pagesize: %ld", slab_pagesize);
+    }
 
     if(unlikely(!ff)) {
         ff = procfile_reopen(ff, PLUGIN_SLABINFO_PROCFILE, " ,:" , PROCFILE_FLAG_DEFAULT);

--- a/collectors/slabinfo.plugin/slabinfo.c
+++ b/collectors/slabinfo.plugin/slabinfo.c
@@ -191,7 +191,7 @@ struct slabinfo *read_file_slabinfo() {
     slabdebug("   Read %lu lines from procfile", (unsigned long)lines);
     for(l = 2; l < lines; l++) {
         if (unlikely(procfile_linewords(ff, l) < 14)) {
-            slabdebug("    Line %zu has only %"PRIu64" words, skipping", l, procfile_linewords(ff,l));
+            slabdebug("    Line %zu has only %zu words, skipping", l, procfile_linewords(ff,l));
             continue;
         }
 

--- a/collectors/slabinfo.plugin/slabinfo.c
+++ b/collectors/slabinfo.plugin/slabinfo.c
@@ -191,7 +191,7 @@ struct slabinfo *read_file_slabinfo() {
     slabdebug("   Read %lu lines from procfile", (unsigned long)lines);
     for(l = 2; l < lines; l++) {
         if (unlikely(procfile_linewords(ff, l) < 14)) {
-            slabdebug("    Line %zu has only %lu words, skipping", l, procfile_linewords(ff,l));
+            slabdebug("    Line %zu has only %"PRIu64" words, skipping", l, procfile_linewords(ff,l));
             continue;
         }
 

